src/App.tsx

```diff
@@ -1,61 +1,116 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/todo';
+import { User } from './types/user';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [listOfTodos, setTodos] = useState(todos);
+  const [newTodoWrite, setNewTodo] = useState('');
+  const [selectUserId, setSelectUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const [todoError, setTodoError] = useState(false);
+
+  const handleFromServer = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTodo = {
+      id: todos[todos.length - 1].id + 1,
```

Comment:
Do you guarantee that the last todo will have the highest index?

Reply:
Agree with Andrii, in case if you'll delete last todo from this list - id will be not relevant
Also we not sure that this list is sorted by id

Not critical in this task, but pay attention to it